### ISSUE TITLE
[Analyzer] - "IDE0017: Object initialization can be simplified" does not trigger on simple using statements

### ISSUE THREAD

**Issue by Rekkonnect** [dotnet/roslyn#77365]
## Version Used

VS 2022 Version 17.13.2
Roslyn 4.13.0 (afe5bc4a827030229ab79ec596b3cb91d4e87872)

Project info
C# 12.0
.NET 8.0

## Steps to Reproduce

Consider the following declaration:
```csharp
public class Disposable(string key, string value) : IDisposable
{
    public string Key { get; set; } = key;
    public string Value { get; set; } = value;
    public string? ExtraValue { get; set; }

    public void Dispose()
    {
    }
}
```

Initializing an object like this will correctly trigger IDE0017:
```csharp
public static void Initializing()
{
    var disposable = new Disposable("a", "b");
    disposable.ExtraValue = "c";
}
```

But using that object in a block-scoped using statement does not trigger the diagnostic:
```csharp
public static void Initializing()
{
    using var disposable = new Disposable("a", "b");
    disposable.ExtraValue = "c";
}
```

There is no semantic reason to avoid recommending this refactor; the object is being initialized after it's been created, and we're still in the same scope before the object gets disposed, so it should be valid to recommend the diagnostic

[SharpLab link](https://sharplab.io/#v2:C4LghgzsA0AmIGoA+ABATARgLAChcoGYACFDANhLSIBUBTKASwDsBzAVQmZdwG9ciBJYqQooALEQ5cAygHsAtrWAALLgAoAlP0F8cg/SQAMRAG5gATkVgMIAB1kQwAIwA2tIgF4iTWgHciACI29o6utGoARGAR0EQRThEaANzaBlbBDs5uAHQAogAewOZgAGpgLgCu7l4RAMYRKXqCAL64rXg4hJSBGaFuaqTGANa0AJ6xg6blVRpEIEQAkkF2mWG8qUIkGMYA0mNEPEQsSklEECdEzZ5EI6ON+l2TZZXuh8fAp+cfl9dmL/eCR7bAD8RAKRVK01eRwuX1O7Q2jwkyxC4S0TQEun07XaQA==) for the above example

## Diagnostic ID

IDE0017: Object initialization can be simplified

**Comment by CyrusNajmabadi** [comment]
This is by design. There are different semantics here. If you have a object initializer then the assignment to the using happens *after* the assignments to properties. So if those throw, you won't dispose.

**Comment by Rekkonnect** [comment]
Right, I forgot how the using statement gets lowered. For reference, with the expanded code example where the property is set after the using statement, if the property setter threw, the object would be properly disposed as it would have been constructed.

It's concerning that a property setter would throw though. In most cases, setting a property won't have you expect an exception being thrown, and thus it would be completely harmless to change between the two forms of code, the object initializer and the expanded form. Citing how the raw string literal fixers also include a "may change semantics" note at the end of the fix text, it could be helpful to add the fixer and simply include that note to hint the user about the potential change in behavior, if any of the properties would throw during the object's initializer.

I understand that this is a low-hanging fruit and has virtually no reason to be provided, and there is a workaround by removing the using keyword temporarily, triggering the code fix and then putting it back in. It's just unintuitive and often times it would not even occur that this is by design for that reason stated above. We tend to forget that properties can throw, and for good reason.

**Comment by CyrusNajmabadi** [comment]
> It's concerning that a property setter would throw though. In most cases, setting a property won't have you expect an exception being thrown

This is not the worrying case. The problematic case is:

`obj.Prop = SomethingThatRarelyThrows();`

This certainly happens, and would now be very broken if we changed the logic.
